libs/fixture-generators/src/generate-election/generate_election.ts

```diff
@@ -0,0 +1,184 @@
+import { ballotMeasureText, seal, words } from './source_text';
+
+const generateId = customAlphabet('0123456789abcdefghijklmnopqrstuvwxyz', 12);
+
+function chooseRandom<T>(array: T[]): T {
+  return assertDefined(array[Math.floor(Math.random() * array.length)]);
+}
+
+function chooseRandomSubset<T>(array: T[]): T[] {
+  const numItems = Math.floor(Math.random() * array.length);
+  const result = new Set<T>();
+  while (result.size < numItems) {
+    result.add(chooseRandom(array));
+  }
+  return Array.from(result);
+}
+
+function randomString(maxLength: number, sourceWords: string[]): string {
+  let result = `${maxLength}:`;
+  while (result.length < maxLength) {
+    result += ` ${chooseRandom(sourceWords)}`;
+  }
+  return result.slice(0, maxLength);
+}
+
+type PartialDeep<T> = T extends object
+  ? {
+      [P in keyof T]?: PartialDeep<T[P]>;
+    }
+  : T;
+
+/**
+ * Generates an election with mock content based on the given parameters.
+ */
+export function generateElection(
+  inputConfig: PartialDeep<GenerateElectionConfig>
+): Election {
+  const config = mergeObjects(
+    defaultConfig as Partial<GenerateElectionConfig>,
+    inputConfig
+  ) as GenerateElectionConfig;
+  const { maxStringLengths } = config;
+
+  function generateDistrict(index: number): District {
+    return {
+      id: `district-${index}` as DistrictId,
+      name: randomString(maxStringLengths.districtName, words.locations),
+    };
+  }
+  const districts = range(0, config.numDistricts).map(generateDistrict);
+
+  function generatePrecinct(index: number): Precinct {
+    return {
+      id: `precinct-${index}`,
+      name: randomString(maxStringLengths.precinctName, words.locations),
+    };
+  }
+  const precincts = range(0, config.numPrecincts).map(generatePrecinct);
+
+  function generateBallotStyle(index: number): BallotStyle {
+    return {
+      id: `ballot-style-${index}`,
+      districts: chooseRandomSubset(districts).map((district) => district.id),
+      precincts: chooseRandomSubset(precincts).map((precinct) => precinct.id),
+    };
+  }
+  const ballotStyles = range(0, config.numBallotStyles).map(
+    generateBallotStyle
+  );
+
+  function generateParty(index: number): Party {
+    const partyStr = ' Party';
+    const name =
+      randomString(
+        maxStringLengths.partyFullName - partyStr.length,
+        words.parties
+      ) + partyStr;
+    return {
+      id: `party-${index}` as PartyId,
+      name,
+      fullName: name,
+      abbrev: name
+        .substring(0, maxStringLengths.partyAbbreviation)
+        .toLocaleUpperCase(),
+    };
+  }
+  const parties = range(0, config.numParties).map(generateParty);
+
+  function generateContest(index: number): AnyContest {
+    const baseContest = {
+      id: `contest-${index}`,
+      districtId: chooseRandom(districts).id,
+      title: randomString(maxStringLengths.contestTitle, words.offices),
+    } as const;
+    if (Math.random() < 0.5) {
```

Comment:
You could just make the config specify numCandidateContests and numYesNoContests too